lib/cacheable/controller.rb

```diff
@@ -36,14 +36,26 @@ def response_cache(key_data=nil, version_data=nil, &block)
         return yield
       end
 
-      handler = Cacheable::ResponseCacheHandler.new(self) do |h|
-        h.key_data       = key_data       || cache_key_data
-        h.version_data   = version_data   || cache_version_data
-        h.block          = block
-        h.cache_store    = Cacheable.cache_store
+      handler = Cacheable::ResponseCacheHandler.new(
+        key_data:              key_data || cache_key_data,
+        version_data:          version_data || cache_version_data,
+        env:                   request.env,
+        cache_age_tolerance:   cache_age_tolerance_in_seconds,
+        serve_unversioned:     serve_unversioned_cacheable_entry?,
+        force_refill_cache:    force_refill_cache?,
+        headers:               response.headers,
+        &block
+      )
+
+      status, _headers, body = handler.run!
+
+      return if request.env['cacheable.miss']
+      case request.env['cacheable.store']
```

Comment:
Is there a better interface than the rack env that would make sense for indicating what the caller should do? Maybe doesn’t matter

Reply:
imo this is fine, we use it for the middleware anyways

Reply:
Yeah, it’s just slightly awkward given ResponseCacheHandler is now a major public interface. It’s still quite reasonable though.

Reply:
That would be nice but the return value needs to conform to the rack spec so it has to return `[status, headers, body]`